options-design/mongo/clientopts/clientopts.go

```diff
@@ -0,0 +1,90 @@
+package clientopts
+
+import (
+	"context"
+	"net"
+	"time"
+)
+
+type ContextDialer interface {
+	DialContext(ctx context.Context, network, address string) (net.Conn, error)
+}
+
+type Option interface {
+	option()
+}
+
+type OptionFunc func(*Client) error
+
+type Client struct{}
+
+type Bundle struct{}
+
+func BundleOptions(...Option) Bundle { return Bundle{} }
+
+func (b *Bundle) Appname(string) *Bundle                                  { return nil }
+func (b *Bundle) AuthMechanism(string) *Bundle                            { return nil }
+func (b *Bundle) AuthMechaismProperties(map[string]string) *Bundle        { return nil }
+func (b *Bundle) AuthSource(s string) *Bundle                             { return nil }
+func (b *Bundle) ConnectTimeout(d time.Duration) *Bundle                  { return nil }
+func (b *Bundle) Dialer(d ContextDialer) *Bundle                          { return nil }
+func (b *Bundle) HeartbeatInterval(d time.Duration) *Bundle               { return nil }
+func (b *Bundle) Hosts(s []string) *Bundle                                { return nil }
+func (b *Bundle) Journal(v bool) *Bundle                                  { return nil }
+func (b *Bundle) LocalThreshold(d time.Duration) *Bundle                  { return nil }
+func (b *Bundle) MaxConnIdleTime(d time.Duration) *Bundle                 { return nil }
+func (b *Bundle) MaxConnsPerHost(u uint16) *Bundle                        { return nil }
+func (b *Bundle) MaxIdleConnsPerHost(u uint16) *Bundle                    { return nil }
+func (b *Bundle) Password(s string) *Bundle                               { return nil }
+func (b *Bundle) ReadConcernLevel(s string) *Bundle                       { return nil }
+func (b *Bundle) ReadPreference(s string) *Bundle                         { return nil }
+func (b *Bundle) ReadPreferenceTagSets(m []map[string]string) *Bundle     { return nil }
+func (b *Bundle) MaxStaleness(d time.Duration) *Bundle                    { return nil }
+func (b *Bundle) ReplicaSet(s string) *Bundle                             { return nil }
+func (b *Bundle) ServerSelectionTimeout(d time.Duration) *Bundle          { return nil }
+func (b *Bundle) Single(v bool) *Bundle                                   { return nil }
+func (b *Bundle) SocketTimeout(d time.Duration) *Bundle                   { return nil }
+func (b *Bundle) SSL(v bool) *Bundle                                      { return nil }
```

Comment:
Can we consolidate SSL options into a single thing?

Reply:
I think for similar reasons to read preference and write concern probably not.